### ISSUE TITLE
Tweaks

### ISSUE THREAD

**Issue by tunnckoCore** [doowb/github-content#1]
Great idea for lib. I also had similar idea started it with http://npm.im/read-source-stream lib.

But okey, few tweaks needed: There's no sense to add options to special `.cache` property, cuz they can be set directly to `.options` property and then they will generate automagically the url because the private [`interpolate`](https://github.com/jonschlinkert/github-base/blob/master/lib/utils.js#L72-L83) from `github-base`.

So if you remove these two parts [#L134-L139](https://github.com/doowb/github-content/blob/master/index.js#L134-L139) and [#L42-L46](https://github.com/doowb/github-content/blob/master/index.js#L42-L46); update `.owner`, `.repo` and `.branch` methods; and just add one merge (merge opts coming from `.file` and `this.options`) in `.file` method all will work.

Cheers.

**Comment by tunnckoCore** [comment]
Wait... just realized. First paragraph is not absolutely correct.
We should just use `this.request` in `.file` method instead of `this.get`

``` js
GithubContent.prototype.file = function(fp, options, cb) {
  if (typeof options === 'function') {
    cb = options;
    options = {};
  }
  options = options || {};
  options = extend(this.options, options, {filepath: fp});

  this.request('get', ':owner/:repo/:branch/:filepath', options, function(err, content) {
    if (err) return cb(err);
    cb(null, {path: fp, content: content});
  });
  return this;
};
```

Huh, hope this work - i'm almost sure, try :)

This will allow to pass different owner/repo for different `.file` calls - currently unavailable.

**Comment by doowb** [comment]
@tunnckoCore thanks for the ideas! I think I put this together using just the `this.cache` object at first and used the methods to make a fluent api. Then I add the `options` to set those things. I don't know why I forgot about using `:` parameters in the `url` instead of building the url dynamically.

I think we can still use `this.get` because it'll do the "right" thing with options passed to it.

**Comment by tunnckoCore** [comment]
Mm yea, anyway it will do the "template thing" between `url` and `options` passed to `.file`, but last sentence is a good feature :)

Okey, but one more thing that I've just noticed, **again**. In the readme examples response object is with `contents` property, but actually it is `content`.

I don't know but I notice this "problem" from few months and it's not for the first time and not only here (it can be seen also in ["render a view" example](https://github.com/assemble/assemble-core#examples) from `assemble-core` - as you can see `.page(str, {content: '..'})` and below `view.contents`) and I think it comes from `layouts` lib (or something around it - i'm not sure, but i did deep research before few months). 

Yea, this can be intentional and I understand partially whats the sense behind plural and singular form using in the different places, but it may be confusing for some people - i definitely dont like it. From few months I strongly want to share this my thought somewhere - i think it should be in plural form everywhere or at least allow accept both forms everywhere. I think we should follow gulp/vinyl way and i dont know why you both with @jonschlinkert choose this way - to have differences.

**Comment by doowb** [comment]
@charlike wow, sorry this took so long... I refactored and removed the fluent part of the api. It's just based on options now and those can be passed in the constructor the methods. I also changed everything to `.contents` since the value type is `Buffer` and it aligns with `vinyl`. Also, the `assemble-core` examples have been updated since this issue was opened.

I'm going to close this now since I think everything is addressed in the refactor.
